### PR TITLE
free_contents should also set the length to zero

### PR DESCRIPTION
**Pull request by iboukris**

### CODE REVIEW

**Comment by greghudson** [comment]
The RPC tests are failing because of https://github.com/krb5/krb5/blob/master/src/lib/gssapi/krb5/util_cksum.c#L101

I'm trying to figure out why the code wants to replace that checksum buffer with a copy of itself; probably some kind of adherence to a no-longer-relevant allocation regimen.

**Comment by iboukris** [comment]
Locally I also have this failure:

```
$ PYTHONPATH=/home/admin/git/krb5/src/util /usr/bin/python3 ./t_iprop.py -v
...
kpropd: Still waiting for full resync
*** Failure: kadmind gave consecutive full resyncs
*** Last mark: propagate M->1 full
*** Last command (#20): /home/admin/git/krb5/src/kprop/kpropd -D -P 61003 -f /home/admin/git/krb5/src/tests/testdir/incoming-datatrans -p /home/admin/git/krb5/src/kadmin/dbutil/kdb5_util -a /home/admin/git/krb5/src/tests/testdir/kpropd-acl -d
*** Output of last command:
ready
*** Failed with db: db2
*** Daemon pid 111574 output:
Invalid iprop header
Generic remote error: incomplete write while writing database block starting at 
offset 0 (4595 written, 0 expected)
```

File: src/kprop/kpropd.c [hunk 1]
```diff
@@ -1401,18 +1401,16 @@ recv_database(krb5_context context, int fd, int database_fd,
         if (krb5_is_krb_error(&inbuf))
             recv_error(context, &inbuf);
         retval = krb5_rd_priv(context, auth_context, &inbuf, &outbuf, NULL);
+        krb5_free_data_contents(context, &inbuf);
```
Comment (frozencemetery) [inline src/kprop/kpropd.c]: This seems out of scope.  Accidentally pulled in?

Comment (iboukris) [reply src/kprop/kpropd.c]: Yeah, technically we just need to move outbuf downward, i'll remove.
